code/modules/mob/mob.dm

```diff
@@ -675,10 +675,12 @@
 			M.LAssailant = usr
 
 	else if(isobj(AM))
-		var/obj/I = AM
-		if(!can_pull_size || can_pull_size < I.w_class)
-			to_chat(src, "<span class='warning'>It won't budge!</span>")
-			return
+		//Let maintenance drones pull disposal pipe
+		if(!(istype(src, /mob/living/silicon/robot/drone) && istype(AM,/obj/structure/disposalconstruct)))

```

Comment:
Terrible, terrible. This is a bad way to make it work. you need to override pulling proc

Reply:
yeah I thought so, but if I redefine the pulling proc, I think I'd need to redefine the whole thing inside of drone.dm except with that change, because like the previous proc I removed from drone.dm, it'd check if it was a disposal pipe, then go on to execute the parent proc inside mob.dm which wouldn't make the same check.

Reply:
What about like this? https://gist.github.com/Baa14453/ecd36628014d3cf875afb02916fcc58d line 330

Reply:
With  comment above, revert this proc back to what it was.